### DISCUSSION TITLE
univer不支持直创建一个只读模式的表格吗？

### DISCUSSION THREAD

**Discussion by LianYiXiang** [dream-num/univer#1771] [Q&A | 常见问题]
我不需要来新增sheet表 也不需要右键 不需要编辑 只需要可以滚动查看 如果可能的话 能两个表一起交互的话更好 左边右边同时滚动

**Comment by jikkai** [comment]
1. 完整的保护区域功能在开发中，请参考 https://univer.ai/zh-cn/guides/introduction/roadmap/
2. 临时的解决方案可以考虑：
  - 阻止命令执行：https://univer.ai/zh-cn/guides/facade/general/#%E9%98%BB%E6%AD%A2%E5%91%BD%E4%BB%A4%E6%89%A7%E8%A1%8C
  - CSS [pointer-events](https://developer.mozilla.org/en-US/docs/Web/CSS/pointer-events) 设置为 none

**Reply by LianYiXiang** [reply]
我看底层univer应该也是封装进去的x-spreadsheet 为什么不直接在创建的时候 继续保持x-spreadsheet的mode 一个edit一个read

**Reply by jikkai** [reply]
Univer 并没有封装 x-spreadsheet

**Reply by Xiangzi529** [reply]
好吧 那可能记错了 我以为是基于spreadsheet开发的了 我还好奇为啥没开发只读模式 不好意思

**Reply by Xiangzi529** [reply]
![image](https://github.com/dream-num/univer/assets/165776161/95839c82-048a-4b5c-8dd8-01b8c1a062f2)
像这种右键菜单 可以定义只显示哪些吗

**Reply by jikkai** [reply]
https://github.com/dream-num/univer/discussions/1705 https://github.com/dream-num/univer/discussions/1642

**Reply by Xiangzi529** [reply]
好的 那我是不是可以先基于css来进行隐藏

**Comment by yooyea** [comment]
pointer-events 设置为 none 后就不能滚动了

**Comment by fmwww** [comment]
可以放开部分操作，屏蔽所有其他操作，实现只读效果，不过这种方式没法屏蔽右键菜单，但是右键菜单里面的操作是没法执行的
```
const enabledOperations = [
  SetSelectionsOperation.id,
  SetWorksheetActiveOperation.id,
  SetWorksheetActivateCommand.id,
  SetFormulaCalculationNotificationMutation.id,
  SetArrayFormulaDataMutation.id,
  SetFormulaCalculationResultMutation.id,
  SetFormulaCalculationStartMutation.id,
  SetFormulaCalculationStopMutation.id,
  SetScrollRelativeCommand.id,
  SetScrollOperation.id,
];

univerAPI.onBeforeCommandExecute((commandInfo) => {
  if (!enabledOperations.includes(commandInfo.id)) {
    throw new Error("sheet is readonly");
  }
});
```

**Reply by Xiangzi529** [reply]
现在有开权限解决的办法 但是我不会配置

**Comment by hexf00** [comment]
In version v0.1.13, we have added an official implementation for sheet permission control. You can find an example here: https://univer.ai/playground?title=sheet+permission . For detailed information, please read: https://univer.ai/guides/sheet/features/permission .

**Comment by zhaoxiong123** [comment]
我看官网的sheet在线demo，为什么protected area还是可以编辑呢？
<img width="634" alt="image" src="https://github.com/dream-num/univer/assets/3273395/fd6e45de-6d53-4a3e-9dbd-5d1fd599dd92">

**Reply by zhaoxiong123** [reply]
已解决，通过pemission point来检查是否有对应权限

**Comment by chad1227** [comment]
有没有办法把protected area 的置灰层格式去掉？保留单元格原来的颜色样式？
